crp_APL/controllers/ctrl_vehicle_control_lqr/scripts/lqr_controller.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python3
+
+import numpy as np
+import scipy.linalg as la
+import time
+import rclpy
+from rclpy.node import Node
+from autoware_control_msgs.msg import Lateral,Longitudinal
+from autoware_planning_msgs.msg import Trajectory
+from crp_msgs.msg import Ego
+
+
+show_animation = True
+
+def angle_mod(x, zero_2_2pi=False, degree=False):
+
+
+    if isinstance(x, float):
+        is_float = True
+    else:
+        is_float = False
+
+    x = np.asarray(x).flatten()
+    if degree:
+        x = np.deg2rad(x)
+
+    if zero_2_2pi:
+        mod_angle = x % (2 * np.pi)
+    else:
+        mod_angle = (x + np.pi) % (2 * np.pi) - np.pi
+
+    if degree:
+        mod_angle = np.rad2deg(mod_angle)
+
+    if is_float:
+        return mod_angle.item()
+    else:
+        return mod_angle
+
+class State:
+
+    def __init__(self, x=0.0, y=0.0, yaw=0.0, v=0.0,steer=0.0):
+        self.x = x
+        self.y = y
+        self.yaw = yaw
+        self.v = v
+        self.steer = steer
+
+def pi_2_pi(angle):
+    return angle_mod(angle)
+
+# create an ROS controller class
+class ROSController(Node):
+
+    def __init__(self):
+
+        super().__init__('lqr_on_rails')
+
+        self.cx = []
+        self.cy = []
+        self.cyaw = []
+        self.ck = []
+        self.sp = []
+
+        self.pe = 0.0
+        self.pth_e = 0.0
+
+        self.state = State(x=0.0, y=0.0, yaw=0.0, v=0.0, steer=0.0)
+
+        self.dt = 0.05
+        self.L = 2.9
+        self.max_steer = np.deg2rad(20.0)
```

Comment:
Max steer as parameter?

Reply:
its only the default, if changed the params from the launch file will overwrite it